projects/dvl-fw/src/lib/plugins/nsf/nsf-validator.ts

```diff
@@ -0,0 +1,36 @@
+import { differenceBy, toLower, trim as loTrim, trimEnd } from 'lodash';
+
+const nsfHeaderSet: Array<string> = Array<string>('AwardNumber', 'Title', 'NSFOrganization', 'Program(s)', 'StartDate', 'LastAmendmentDate',
+'PrincipalInvestigator', 'PI', 'AllPI', 'State', 'Organization', 'AwardInstrument', 'ProgramManager', 'EndDate', 'AwardedAmountToDate',
+'Co-PIName(s)', 'PIEmailAddress', 'OrganizationStreet', 'OrganizationCity', 'OrganizationState', 'OrganizationZip', 'OrganizationPhone',
+'NSFDirectorate', 'ProgramElementCode(s)', 'ProgramReferenceCode(s)', 'ARRAAmount', 'Abstract');
+const tolerance = 0; // %
```

Comment:
Just make tolerance be the number of fields missing (ie they can have at most 2 fields missing).